### ISSUE TITLE
Improve error-handling

### ISSUE THREAD

**Issue by meejah** [LeastAuthority/magic-folder#378]
For example: https://github.com/LeastAuthority/magic-folder/blob/main/src/magic_folder/uploader.py#L280

That comment is wrong and catching `Exception` is too general. There are lots of reasons the called code may fail, not only "tahoe isn't working". I'm picking on that example but there are other places where exceptions are incorrectly eaten/ignored so things like "syntax error" cause code to silently fail.

Another spot: https://github.com/LeastAuthority/magic-folder/blob/main/src/magic_folder/uploader.py#L172

Many of the IServices which follow the pattern of a LoopingCall to find more work also don't log errors correctly (although they should at least continue to function on the next loop) and should also be fixed.

**Comment by tomprince** [comment]
> Many of the IServices which follow the pattern of a LoopingCall to find more work also don't log errors correctly (although they should at least continue to function on the next loop) and should also be fixed.

I suspect it may make sense to convert these to use `TimerService`, instead of managing the `LoopingCall` manually. This is distinct from handling errors, but we could use a wrapper that added some error handling around the called function.

**Comment by meejah** [comment]
Yeah looks like https://twistedmatrix.com/documents/20.3.0/api/twisted.application.internet.TimerService.html could replace several of our "work queue"-type services.
